rules/CodingStyle/Application/UseImportsAdder.php

```diff
@@ -253,4 +291,76 @@ private function isCurrentNamespace(string $namespaceName, ObjectType $objectTyp
 
         return ! \str_contains($afterCurrentNamespace, '\\');
     }
+
+    /**
+     * @param Stmt[] $stmts
+     * @param Use_[] $newUses
+     * @return Stmt[]
+     */
+    private function insertUseNodesInStatements(
+        array $stmts,
+        array $newUses,
+        int $position = 0,
+        bool $addSpace = true
+    ): array {
+        $importInsertSorted = SimpleParameterProvider::provideBoolParameter(Option::IMPORT_INSERT_SORTED);
+        if ($importInsertSorted && isset($stmts[$position])
+            && ($stmts[$position] instanceof Use_ || $stmts[$position] instanceof GroupUse)) {
+            foreach ($newUses as $newUse) {
+                do {
+                    $useListPosition = 0;
+                    $prefix = '';
+                    if (! isset($stmts[$position])
+                        || (! $stmts[$position] instanceof Use_ && ! $stmts[$position] instanceof GroupUse)) {
+                        break;
+                    }
+
+                    if ($stmts[$position] instanceof GroupUse) {
+                        $prefix = $stmts[$position]->prefix->toString() . '\\';
+                    }
+
+                    $parentUseType = $stmts[$position]->type;
+                    $newUseType = $newUse->type;
+                    foreach ($stmts[$position]->uses as $use) {
+                        $currentUseType = $parentUseType === Use_::TYPE_UNKNOWN ? $use->type : $parentUseType;
```

Comment:
Find out the use type of the current `UseUse`. In most cases it will just be the type of the parent `Use_` statement. The exception will be mixed group use types. In that case the type of the parent will be "Unknown" and we will have to use the type saved in the `UseUse` node